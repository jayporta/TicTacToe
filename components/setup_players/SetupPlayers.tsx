import React, { useState } from 'react';
import { TextInput, Text, View, TouchableOpacity } from 'react-native';
import { Players } from '../../types';
import styles from './setupPlayersStyles';

const SetupPlayers = ({
  onSavePlayers,
  existingPlayers,
}: {
  onSavePlayers: (players: Players) => void;
  existingPlayers: Players;
}) => {
  const [players, setPlayers] = useState(existingPlayers);
  const playersValidated = players.X && players.O;

  const handleSubmit = () => (playersValidated ? onSavePlayers(players) : null);

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Enter Player Names</Text>
      {Object.keys(players).map((key, i) => (
        <View key={i} style={styles.player}>
          <Text>{`Player ${key}: `}</Text>
          <TextInput
            accessibilityHint={`Enter name for Player ${key}`}
            autoCorrect={false}
            autoCompleteType="off"
            autoFocus={i === 0}
            style={styles.playerInput}
            value={players[key as keyof Players]}
            onChangeText={input =>
              setPlayers({ ...players, [key as keyof Players]: input })
            }
            onSubmitEditing={handleSubmit}
            returnKeyType="go"
            placeholder="Enter name"
          />
        </View>
      ))}
      <TouchableOpacity
        accessibilityRole="button"
        accessibilityHint="Tap to play"
        style={styles.saveButton}
        onPress={handleSubmit}
      >
        <Text style={styles.saveButtonText}>Go</Text>
      </TouchableOpacity>
    </View>
  );
};

export default SetupPlayers;
