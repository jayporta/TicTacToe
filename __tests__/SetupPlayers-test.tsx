import 'react-native';
import React from 'react';
import SetupPlayers from '../components/setup_players/SetupPlayers';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('SetupPlayers matches snapshot', () => {
  const onSaveMock = jest.fn();
  const tree = renderer
    .create(
      <SetupPlayers
        onSavePlayers={onSaveMock}
        existingPlayers={{ X: '', O: '' }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
