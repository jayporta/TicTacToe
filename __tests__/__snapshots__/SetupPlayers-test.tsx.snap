// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetupPlayers matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginTop": 70,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 30,
        "marginBottom": 50,
        "textAlign": "center",
      }
    }
  >
    Enter Player Names
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderStyle": "solid",
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "fontSize": 18,
        "height": 40,
        "marginBottom": 30,
        "paddingLeft": 5,
        "width": 300,
      }
    }
  >
    <Text>
      Player X: 
    </Text>
    <TextInput
      accessibilityHint="Enter name for Player X"
      allowFontScaling={true}
      autoCompleteType="off"
      autoCorrect={false}
      autoFocus={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Enter name"
      rejectResponderTermination={true}
      returnKeyType="go"
      style={
        Object {
          "flex": 1,
          "fontSize": 18,
          "fontWeight": "500",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderStyle": "solid",
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "fontSize": 18,
        "height": 40,
        "marginBottom": 30,
        "paddingLeft": 5,
        "width": 300,
      }
    }
  >
    <Text>
      Player O: 
    </Text>
    <TextInput
      accessibilityHint="Enter name for Player O"
      allowFontScaling={true}
      autoCompleteType="off"
      autoCorrect={false}
      autoFocus={false}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Enter name"
      rejectResponderTermination={true}
      returnKeyType="go"
      style={
        Object {
          "flex": 1,
          "fontSize": 18,
          "fontWeight": "500",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    accessibilityHint="Tap to play"
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "green",
        "borderRadius": 10,
        "borderWidth": 1,
        "opacity": 1,
        "paddingVertical": 12,
        "width": 200,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Go
    </Text>
  </View>
</View>
`;
